### PR TITLE
Create dependent records and appeals for CAVC appeals FactoryBot

### PR DESCRIPTION
**Pull request by yoomlam**
Bumps [CASEFLOW-93](https://vajira.max.gov/browse/CASEFLOW-93) and Part 2 of #15774.

### Description
Make `cavc_remand` FactoryBot for robust.

An appeal resulting from a CAVC Remand depends on the `CavcRemand` record to exist so that the appropriate tasks can be created depending on the `CavcRemand.subtype`. It also relies on records about the claimant, veteran, decision issues, and decision documents of the source appeal of the `CavcRemand`.

When creating a CAVC Remand appeal using FactoryBot, the factory should allow for the dependencies to be passed in or created as needed. In the end, all the records should be consistent with each other, at least enough to be used in an RSpec test. 

### Acceptance Criteria
- [x] Code compiles correctly

### CODE REVIEW

File: spec/workflows/bulk_task_assignment_spec.rb [hunk 1]
```diff
@@ -178,19 +178,14 @@ def create_no_show_hearing_task_for_appeal(appeal)
...
-            cavc_aod_appeal,
-            cavc_aod_legacy_appeal,
-            aod_appeal,
-            aod_legacy_appeal,
-            cavc_appeal,
-            cavc_legacy_appeal
-          ]
+          [cavc_aod_appeal, cavc_aod_legacy_appeal].sort_by(&:docket_number) +
+            [aod_appeal, aod_legacy_appeal].sort_by(&:docket_number) +
+            [cavc_appeal, cavc_legacy_appeal].sort_by(&:docket_number)
```
Comment (yoomlam) [inline spec/workflows/bulk_task_assignment_spec.rb:183]: After sorting by AOD and CAVC status, the list should have been sorted by `docket_number`, based on https://github.com/department-of-veterans-affairs/caseflow/blob/2953f4a90de34cee1af6ad4b0a80a7c3bd63a6ea/app/models/task.rb#L221

File: spec/jobs/push_priority_appeals_to_judges_job_spec.rb [hunk 2]
```diff
@@ -221,10 +221,10 @@ def to_judge_hash(arr)
       end
     end
 
-    let(:priority_count) { Appeal.count + VACOLS::Case.count }
+    let(:priority_count) { Appeal.count { |a| a.aod? || a.cavc? } + VACOLS::Case.count }
     let(:priority_target) { (priority_count + judge_distributions_this_month.sum) / judges.count }
 
-    it "should distibute ready priortiy appeals to the judges" do
+    it "should distibute ready priority appeals to the judges" do
```
Comment (n1zyy) [inline spec/jobs/push_priority_appeals_to_judges_job_spec.rb:227]: Found another one! 😄 

```suggestion
    it "should distribute ready priority appeals to the judges" do
```

File: spec/models/tasks/cavc_correspondence_mail_task_spec.rb [hunk 3]
```diff
@@ -153,20 +153,18 @@
...
 
     context "on a CAVC Appeal Stream" do
-      let(:appeal) { create(:appeal, :type_cavc_remand, :with_post_intake_tasks) }
+      let(:appeal) { create(:appeal, :type_cavc_remand) }
+      let(:root_task) { appeal.root_task }
 
       context "without a CAVC task" do
-        let(:appeal) { create(:appeal, :type_cavc_remand) }
-        let(:root_task) { RootTask.create!(appeal: appeal) }
+        before { CavcTask.find_by(appeal: appeal).destroy }
```
[the remark this answers is not in the document]
Comment (n1zyy) [reply spec/models/tasks/cavc_correspondence_mail_task_spec.rb:160]: Is a CAVC appeal stream without a CavcTask now considered bad data? I.e., is this testing a case we still care about?

File: spec/factories/appeal.rb [hunk 4]
```diff
@@ -265,8 +287,9 @@
     trait :assigned_to_judge do
       ready_for_distribution
       after(:create) do |appeal, evaluator|
+        root_task = RootTask.find_or_create_by!(appeal: appeal, assigned_to: Bva.singleton)
```
[the remark this answers is not in the document]
Comment (n1zyy) [reply spec/factories/appeal.rb:287]: Is that a valid condition, or is this papering over deficiencies in those tests?

**Review by n1zyy — COMMENTED** [review]
This looks good overall, but a couple questions to make sure I'm understanding everything correctly.

File: spec/factories/appeal.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (yoomlam) [reply spec/factories/appeal.rb:287]: The idea of FactoryBot is to create everything needed so the tester doesn't have to remember to do it. The tester can create their own RootTask if it is relevant to their test, but they don't have to.

File: spec/models/tasks/cavc_correspondence_mail_task_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (yoomlam) [reply spec/models/tasks/cavc_correspondence_mail_task_spec.rb:160]: In the CAVC reversal workflow, we decided (today) that a CavcTask is not required for a CAVC appeal, so it is not considered bad data.

[the remark this answers is not in the document]
Comment (n1zyy) [reply spec/models/tasks/cavc_correspondence_mail_task_spec.rb:160]: Oh! I hadn't connected those dots earlier. 👍

**Review by n1zyy — APPROVED** [review]
LGTM!

**Pull request merged**